account_reconcile_compassion/models/bank_statement.py

```diff
@@ -38,3 +39,88 @@ def _default_name(self):
+                """
+        params = (self.env.user.company_id.id,)
+        if statements:
+            sql_query += ' AND stl.statement_id IN %s'
+            params += (tuple(statements.ids),)
+        sql_query += ' ORDER BY partner.lastname, partner.firstname, ' \
+                     'partner.id'
+        self.env.cr.execute(sql_query, params)
+        x = self.env.cr.dictfetchall()
+        lines_left = self.env['account.bank.statement.line'].browse([
+            line.get('id') for line in x])
+        # try to assign partner to bank_statement_line
+        stl_to_assign_partner = [stl.id for stl in lines_left if not
+                                 stl.partner_id]
+        refs = list(set([st.name for st in lines_left if not
+                         stl.partner_id]))
+        if lines_left and stl_to_assign_partner and refs \
+                and lines_left[0].journal_id.default_credit_account_id \
+                and lines_left[0].journal_id.default_debit_account_id:
+
+            sql_query = """SELECT aml.partner_id, aml.ref, stl.id
+                            FROM account_move_line aml
+                                JOIN account_account acc ON acc.id =
+                                aml.account_id
+                                JOIN account_bank_statement_line stl ON
+                                aml.ref = stl.name
+                                JOIN res_partner AS partner ON partner.id =
+                        stl.partner_id
+                            WHERE (aml.company_id = %s
+                                AND aml.partner_id IS NOT NULL)
+                                AND (
+                                    (aml.statement_id IS NULL AND
+                                    aml.account_id IN %s)
+                                    OR
+                                    (acc.internal_type IN ('payable',
+                                    'receivable') AND aml.reconciled = false)
+                                    )
+                                AND aml.ref IN %s
+                                """
+            params = (self.env.user.company_id.id,
+                      (lines_left[0].journal_id.default_credit_account_id.id,
+                       lines_left[0].journal_id.default_debit_account_id.id),
+                      tuple(refs))
+            if statements:
+                sql_query += 'AND stl.id IN %s'
+                params += (tuple(stl_to_assign_partner),)
+            sql_query += ' ORDER BY partner.lastname, partner.firstname, ' \
+                         'partner.id'
+            self.env.cr.execute(sql_query, params)
+            results = self.env.cr.dictfetchall()
+            st_line = self.env['account.bank.statement.line']
+            for line in results:
+                st_line.browse(line.get('id')).write(
+                    {'partner_id': line.get('partner_id')})
+        return {
+            'st_lines_ids': lines_left.ids,
```

Comment:
Could you just instead use the super method in order to avoid copying all the code? This way if something changes, we don't override everything. I think of something like :
```
result = super(..., self).reconciliation_widget_preprocess()
line_ids = result['st_lines_ids']
... reorder process
result['st_lines_ids'] = new_order_ids
return result
```

Reply:
Missed this opportunity. Should be much better now !